File: solutions/beecrowd/2140/2140.cpp
#include <cstdint>
#include <cstring>
#include <iostream>

int main() {
    std::int16_t n, m;
    std::int16_t bills[] = {2, 5, 10, 20, 50, 100};
    std::int16_t possible_changes[201];

    memset(possible_changes, 0, sizeof(possible_changes));

    for (std::int16_t i = 0; i <= 5; i++) {
        for (std::int16_t j = 0; j <= 5; j++) {
            possible_changes[bills[i] + bills[j]] = bills[i] + bills[j];
        }
    }

    while (std::cin >> n >> m && n && m) {
        if (m - n <= 200 && possible_changes[m - n]) {
            std::cout << "possible" << std::endl;
        } else {
            std::cout << "impossible" << std::endl;
        }
    }

    return 0;
}

File: solutions/beecrowd/2140/2140.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

bills = [2, 5, 10, 20, 50, 100]
possible_changes = [0] * 201

for bill_1 in bills:
    for bill_2 in bills:
        possible_changes[bill_1 + bill_2] = bill_1 + bill_2

while line := input().strip():
    n, m = map(int, line.split())

    if n == 0 and m == 0:
        break

    if m - n in possible_changes and possible_changes[m - n]:
        output('possible\n')
    else:
        output('impossible\n')
